src/main/scala/TypeCheck.scala

```diff
@@ -63,6 +63,9 @@ object TypeChecker {
           env2.update(id -> env2(id).consumeDim(i, e - s)) -> bres * (e - s)
         case (TStaticInt(v), env2) =>
           env2.update(id -> env(id).consumeBank(i, v % dims(i)._2)) -> bres * 1
+        case (TSizedInt(_), env2) =>
+          if (dims(i)._2 != 1) throw BankUnrollInvalid(dims(i)._2, 1) //different error?
```

Comment:
Add a new error called `InvalidDynamicIndex` based on the `InvalidIndex` error message and explain the conditions required to index with a dynamic int.